tests/unit/assessment/test_assessment.py

```diff
@@ -531,6 +531,70 @@ def test_pipeline_list_with_no_config():
     assert len(crawler) == 0
 
 
+def test_cluster_assessment_with_spn_cluster_policy_not_found(mocker):
+    sample_clusters = [
+        ClusterDetails(
+            autoscale=AutoScale(min_workers=1, max_workers=6),
+            cluster_source=ClusterSource.UI,
+            spark_context_id=5134472582179565315,
+            spark_env_vars=None,
+            spark_version="9.3.x-cpu-ml-scala2.12",
+            cluster_id="0810-225833-atlanta69",
+            cluster_name="Tech Summit FY24 Cluster-1",
+            policy_id="bdqwbdqiwd1111",
+        )
+    ]
+    ws = mocker.Mock()
+    ws.clusters.list.return_value = sample_clusters
+    ws.cluster_policies.get.side_effect = DatabricksError(error="NO_POLICY", error_code="NO_POLICY")
```

Comment:
Side effect has to raise error, not return it

Reply:
Please elaborate on what do you mean

Reply:
please elaborate on what you mean. This is testing a scenario where a policy does not exist which means the ws.cluster_policies.get method throws an exception.